tests/pems/ocsp/OCSP-TEST.md

```diff
@@ -21,11 +21,20 @@ The CN for this cert matches the URI in the Server Cert's "Authority Information
 
 The leaf cert/key. OCSP responses will be generated for this cert.
 
+### Early Expiry Cert
+The `server_cert_early_expire.pem` is a certificate that expires in 2037. This certificate is used to validate correct behavior of expired dates for unit tests on 32 bit platforms. This is necessary because 32 bit `time_t` values can't represent dates past 2038, so our unit tests on 32 bit platforms can't test the expiry of very long lived certs.
+
+This cert and the corresponding ocsp response, `ocsp_response_early_expire.der` can be generated at any time by running the `./generate.sh` script. This will break the `s2n_x509_validator_test_offset` unit test, because it will change the `this_update` time of the `ocsp_reponse_early_expire.der`.
```

Comment:
Probably not that big of a concern since we shouldn't really have to regenerate the OCSP response, but should the this_update time be fixed like the next_update time is, so the test doesn't break?

Reply:
I would love if that were the case, but couldn't find any way to do it. the `-attime` parameter was the only one that dealt with "current time" but that didn't seem to fix the "this-update" field.

The only other way I could think of doing this was disabling ntp, setting the system time to the expected time, and then re-enabling ntp at the end of cert generation, which seemed very fragile.

Reply:
Ah ok yeah... It looks hardcoded to the current time: https://github.com/openssl/openssl/blob/4f373a9773efa63fdb73f3972f13ab78b9342b70/apps/ocsp.c#L1058

Reply:
Could an alternative potentially be to use the libcrypto to read the this_update/next_update times off of the OCSP response within the test?

Reply:
I did think about doing that, but spent some time looking at OpenSSL api's and am quite convinced that the amount of time it would take me to understand the stuffer things/file read into the stuff/call the right openssl apis is a larger amount of time than someone will ever spend updating this information, considering the last time new OCSP certificates was introduced was 6 years ago.